src/Core/Parser/TemplateProcessor/NamespaceDetectionTemplateProcessor.php

```diff
@@ -103,18 +102,20 @@ public function registerNamespacesFromTemplateSource($templateSource)
         if ($matched) {
             $namespaces = [];
             preg_match_all('/' . $namespacePattern . '/', $matches[0], $namespaces, PREG_SET_ORDER);
-            foreach ($namespaces as $set) {
-                $namespaceUrl = trim($set[2], '"\'');
-                if (strpos($namespaceUrl, 'http://typo3.org/ns/') === 0) {
-                    $namespaceUri = substr($namespaceUrl, 20);
-                    $namespacePhp = str_replace('/', '\\', $namespaceUri);
-                } elseif (!preg_match('/([^a-z0-9_\\\\]+)/i', $namespaceUrl)) {
-                    $namespacePhp = $namespaceUrl;
-                    $namespacePhp = preg_replace('/\\\\{2,}/', '\\', $namespacePhp);
-                } else {
-                    $namespacePhp = null;
+            if (!empty($namespaces)) {
```

Comment:
This can be dropped, an empty array will not be iterated over anyways.

Reply:
The point is two-fold: to skip empty arrays *and* skip `false`, while avoiding use of `is_array` (which admittedly is slightly opinionated but, that's my chosen way of making a potential future switch to `iterable` less painful).